packages/filebrowser/src/listing.ts

```diff
@@ -2237,12 +2245,32 @@ export namespace DirListing {
         node.removeAttribute('data-is-dot');
       }
       // If an item is being edited currently, its text node is unavailable.
+      const indices = !model.indices ? [] : model.indices;
+      let highlightedName = StringExt.highlight(model.name, indices, h.mark);
       if (text) {
-        const indices = !model.indices ? [] : model.indices;
-        let highlightedName = StringExt.highlight(model.name, indices, h.mark);
         VirtualDOM.render(h.span(highlightedName), text);
       }
 
+      // Adds an aria-label to the checkbox element.
+      const checkbox = checkboxWrapper?.querySelector(
+        'input[type="checkbox"]'
+      ) as HTMLInputElement;
+
+      if (checkbox) {
+        let ariaLabel: string;
+        if (fileType.contentType === 'directory') {
+          ariaLabel = selected
+            ? trans.__('Unselect directory "%2"', highlightedName)
+            : trans.__('Select directory "%2"', highlightedName);
+        } else {
+          ariaLabel = selected
+            ? trans.__('Unselect file "%2"', highlightedName)
+            : trans.__('Select file "%2"', highlightedName);
```

Comment:
```suggestion
            ? trans.__('Unselect directory "%1"', highlightedName)
            : trans.__('Select directory "%1"', highlightedName);
        } else {
          ariaLabel = selected
            ? trans.__('Unselect file "%1"', highlightedName)
            : trans.__('Select file "%1"', highlightedName);
```

Currently it shows:

![Screenshot from 2023-01-04 22-58-11](https://user-images.githubusercontent.com/5832902/210665800-ee19bc53-3605-49cc-85d1-65c164d47ee3.png)

Reply:
might be worth adding a test to [packages/filebrowser/test/listing.spec.ts](https://github.com/jupyterlab/jupyterlab/blob/master/packages/filebrowser/test/listing.spec.ts)

Reply:
Sorry for this leftover, I did it too quickly.